src/wallet/wallet.cpp

```diff
@@ -37,7 +37,36 @@
 #include <boost/algorithm/string/replace.hpp>
 #include <boost/thread.hpp>
 
-std::vector<CWalletRef> vpwallets;
+static std::vector<std::shared_ptr<CWallet>> vpwallets;
+
+bool AddWallet(std::shared_ptr<CWallet> pwallet)
```

Comment:
These all need locking.

Reply:
At the moment no because wallets are added at startup. This is mentioned in the PR description:
> make the new interface thread safe.